### PR TITLE
test: add the unit test and comments of math_sphbes.h 

### PR DESCRIPTION
**Pull request by pxlxingliang**

### CODE REVIEW

File: source/module_base/math_sphbes.h [hunk 1]
```diff
@@ -14,26 +14,53 @@ class Sphbes
...
+        const double &q,
+        const int &l,
+        double *jl
     );
 
+    /**
+     * @brief spherical bessel
+     * 
+     * @param msh [in] number of grid points
+     * @param r [in] radial grid (1:msh)
```
Comment (caic99) [inline source/module_base/math_sphbes.h]: Sorry but I can't get the meaning of `(1:msh)` here. Could you explain it to me?

Comment (pxlxingliang) [reply source/module_base/math_sphbes.h]: the index is 1 to msh, may be this is redundant. I think it is better to delete it.

File: source/module_base/test/math_sphbes_test.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+{
+    int neign = 100;
+    double **eign = new double*[8];
+    for(int i=0;i<8;++i)
+    {
+        eign[i] = new double[neign];
+        ModuleBase::Sphbes::Spherical_Bessel_Roots(neign,i,1.0e-12,eign[i],10.0);
+    }
+
+    EXPECT_NEAR(eign[0][0]/0.31415926535899563188,       1.0,doublethreshold);
```
Comment (caic99) [inline source/module_base/test/math_sphbes_test.cpp:139]: Why are you using expect_near with (calculated/expected,1), rather than (calculated, expected)?

Comment (pxlxingliang) [reply source/module_base/test/math_sphbes_test.cpp:139]: I have a series of testes, and the values may have different orders of magnitude, so it is not convenient to use a same threshold. While, by using this type, I can use the same threshold.

Comment (caic99) [reply source/module_base/test/math_sphbes_test.cpp:139]: > I have a series of testes, and the values may have different orders of magnitude, so it is not convenient to use a same threshold. While, by using this type, I can use the same threshold.

OK, I know. Personally, I prefer the way of `expect_near(calculated,expected, expected * error)`. It's all up to you. It's also good to define an macro of something like `expect_near_rel`.

Comment (pxlxingliang) [reply source/module_base/test/math_sphbes_test.cpp:139]: Ok, I think the current form is good.

**Review by caic99 — APPROVED** [review]

**Pull request merged**